packages/core/src/agent/Agent.ts

```diff
@@ -171,14 +172,45 @@ export class Agent {
     // Because this requires the connections module, we do this in the agent constructor
     if (mediatorConnectionsInvite) {
       // Assumption: processInvitation is a URL-encoded invitation
-      let connectionRecord = await this.connections.receiveInvitationFromUrl(mediatorConnectionsInvite, {
-        autoAcceptConnection: true,
-      })
-
-      // TODO: add timeout to returnWhenIsConnected
-      connectionRecord = await this.connections.returnWhenIsConnected(connectionRecord.id)
-      const mediationRecord = await this.mediationRecipient.requestAndAwaitGrant(connectionRecord, 60000) // TODO: put timeout as a config parameter
-      await this.mediationRecipient.setDefaultMediator(mediationRecord)
+      const invitation = await ConnectionInvitationMessage.fromUrl(mediatorConnectionsInvite)
+      // Check if invitation has been used already
+      const connections = await this.connections.getAll()
```

Comment:
Retrieving all connections is not a very scalable solution. We already store the invitation key, so I think we should be able to do the following:

```ts
const invitationKey = invitation.recipientKeys[0]
const connection = await connectionRepository.findSingleByQuery(invitationKey)

if (connection) {
  // ...
}
```

I think this should be a method exposed from the mediator or connection module, so we don't have to interact with connection repository directly

Reply:
I see. I will incorporate this.